lib/pure/oids.nim

```diff
@@ -66,21 +67,23 @@ proc oidToString*(oid: Oid, str: cstring) =
   str[24] = '\0'
 
 proc `$`*(oid: Oid): string =
+  ## Converts an oid to string.
   result = newString(24)
   oidToString(oid, result)
 
-proc rand(): cint {.importc: "rand", header: "<stdlib.h>", nodecl.}
-proc srand(seed: cint) {.importc: "srand", header: "<stdlib.h>", nodecl.}
-
-var t = getTime().toUnix.int32
-srand(t)
 
 var
-  incr: int = rand()
-  fuzz: int32 = rand()
+  t = getTime().toUnix.int32
+  seed = initRand(t)
+  incr: int = seed.rand(0x7fff)
+
+let fuzz = int32(seed.rand(high(int32)))
```

Comment:
`let fuzz = cast[int32](seed.rand(high(int)))` otherwise fuzz is always `>=0` (ie less entropy)